### DISCUSSION TITLE
loading scripts into filesystem during build time?

### DISCUSSION THREAD

**Discussion by brad-arnett** [micropython/micropython#13334] [RP2040 / Pico]
I have a build process set up in a docker container to build micropython from source.  During this process I am freezing modules into the build.  These work as expected.

But is there a way to bake files into the filesystem?  I can use 'ampy' or some other tool to put the files there, but I want them automatically incorporated into the micropython release itself.

**Comment by andrewleech** [comment]
There's no built in way currently, though it is planned with vfsmap https://github.com/micropython/micropython/pull/8381

However it's possible to convert a binary file into a byte string in a python file that can then be frozen into code. Your app can then just import that module and use the byte string like a read file.

**Reply by ma261065** [reply]
Could you back up the vfs partition with your files in it and then just flash that to your device along with the firmware?

See https://github.com/orgs/micropython/discussions/12223

**Reply by ukicomputers** [reply]
Does anyone know how to change boot.py?

**Reply by bixb922** [reply]
You copy boot.py to the root of the file system, just as you copy main.py

**Comment by bixb922** [comment]
This what I did to bake in non-python files into a image:

https://github.com/bixb922/freezefs

This takes a folder with files and subfolders and generates a self-mountable .py file that is frozen together with the rest of the python files. Import the file in boot.py or main.py and the  freezefs file system is mounted.

**Comment by Josverl** [comment]
You can use vfs merge to create and add/include a file system with contents into a firmware image.
It has zero overhead to the image. 

https://github.com/Josverl/vfs_merge

That is the same as if you would copy these files to a file system after flashing, but different from freezing .(m)py files
